components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -824,15 +832,19 @@ public static List<OperationPolicy> findOrImportPolicy(List<OperationPolicy> pol
                         }
                         operationPolicyData.setMd5Hash(
                                 APIUtil.getHashOfOperationPolicy(operationPolicyData));
-                        policyID = provider.importOperationPolicy(operationPolicyData, tenantDomain);
+                        policyID = provider.importOperationPolicyOfGivenType(operationPolicyData,
+                                policyType, tenantDomain);
                         importedPolicies.put(policyFileName, policyID);
                         policyImported = true;
                     } else {
                         // Check whether the policy has been referenced
-                        OperationPolicyData policyData =
-                                provider.getAPISpecificOperationPolicyByPolicyName(policy.getPolicyName(),
-                                        policy.getPolicyVersion(), api.getUuid(), null,
-                                        tenantDomain, false);
+                        OperationPolicyData policyData = null;
+                        if (policyType == null
+                                || ImportExportConstants.POLICY_TYPE_API.equalsIgnoreCase(policyType)) {
+                            policyData = provider.getAPISpecificOperationPolicyByPolicyName(policy.getPolicyName(),
+                                    policy.getPolicyVersion(), api.getUuid(), null,
+                                    tenantDomain, false);
+                        }
```

Comment:
Please add a comment to explain why we do this null or 'api' check.

Reply:
this method is used to validate the provided policy list and return the validated list. if the policy spec is not there, then it checks whether the policy has been referenced. if the policy type is not defined or defined as an api level policy, it fetches the policy data. If the policyData is null , which means there's no API Specific policy with the provided details. Then it's checked within common policies.